### PR TITLE
Private Files: Restrict cross-site file access

### PR DESCRIPTION
**Pull request by mjangda**
In a multisite install, we support running mixed private and public sites. To harden our ACL endpoint and prevent accidentally leaking private files via public sites, we now restrict access to files on the same subsite.

So, `example.com` (main site) can access `example.com/wp-content/uploads/2021/02/donuts.jpg`. But it cannot access `example.com/wp-content/uploads/sites/2/2021/02/icecream.jpg` since that file is managed by Site ID 2.

Similarly, `example.com/en/` (subsite) can access `example.com/en/wp-content/uploads/sites/2/2021/02/icecream.jpg` but not `example.com/wp-content/uploads/2021/02/donuts.jpg`.

### Private Files: Prevent cross-site lookup

In a multisite install, we support running mixed private and public sites. To harden our ACL endpoint and prevent accidentally leaking private files via public sites, we now restrict access to files on the same subsite.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- n/a This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples. 

## Steps to Test

1. `vip-init.sh` on mu-dev
2. `curl http://vip-go-dev.lndo.site/wp-content/mu-plugins/files/acl/endpoint-check-file-acl.php -H 'X-Original-Uri: /wp-content/uploads/fish.jpg' -I` <= should return 202
3. `lando wp core multisite-convert`
4. `lando wp site create --slug=test`
5. 4. `lando wp site create --slug=test2`
5. main site => self: `curl http://vip-go-dev.lndo.site/wp-content/mu-plugins/files/acl/endpoint-check-file-acl.php -H 'X-Original-Uri: /wp-content/uploads/fish.jpg' -I` <= returns 202
6. main site => subsite: `curl http://vip-go-dev.lndo.site/wp-content/mu-plugins/files/acl/endpoint-check-file-acl.php -H 'X-Original-Uri: /wp-content/uploads/sites/2/fish.jpg' -I` <= returns 403
7. sub site => self: `curl http://vip-go-dev.lndo.site/wp-content/mu-plugins/files/acl/endpoint-check-file-acl.php -H 'X-Original-Uri: /test/wp-content/uploads/sites/2/fish.jpg' -I` <= returns 202
8. sub site => main: `curl http://vip-go-dev.lndo.site/wp-content/mu-plugins/files/acl/endpoint-check-file-acl.php -H 'X-Original-Uri: /test/wp-content/uploads/fish.jpg' -I` <= returns 403
9. sub site => another subsite: `curl http://vip-go-dev.lndo.site/wp-content/mu-plugins/files/acl/endpoint-check-file-acl.php -H 'X-Original-Uri: /test2/wp-content/uploads/sites/3/fish.jpg' -I` <= returns 403

### CODE REVIEW

**Review by rinatkhaziev — DISMISSED** [review]
The code looks good, unless @alexcriss has any objections we can ship it.

**Comment by rinatkhaziev** [comment]
Keeping the Needs Review label until Ale chimes in.

File: files/acl/acl.php [hunk 1]
```diff
@@ -54,6 +54,30 @@ function get_option_as_bool( $option_name, $default = false ) {
 	], true );
 }
 
+/**
+ * Check if the path is allowed for the current context.
+ *
+ * @param string $file_path Path to the file, including the `/wp-content/uploads/` bit.
```
Comment (mjangda) [inline files/acl/acl.php:60]:
```suggestion
 * @param string $file_path Path to the file, minus the `/wp-content/uploads/` bit. It's the second portion returned by `Pre_Wp_Utils\prepare_request()`
```

**Comment by alexcriss** [comment]
This looks very nice to me ❤️

**Review by alexcriss — APPROVED** [review]

**Comment by rinatkhaziev** [comment]
Merged a small comment update which dismissed the review - will use admin powers to merge.

**Review by rinatkhaziev — APPROVED** [review]

**Comment by rinatkhaziev** [comment]
r1381-stacks

**Pull request merged**